internal/server/destinationcredentials_test.go

```diff
@@ -0,0 +1,105 @@
+package server
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"sync"
+	"testing"
+	"time"
+
+	"gotest.tools/v3/assert"
+
+	"github.com/infrahq/infra/api"
+	"github.com/infrahq/infra/internal/server/data"
+	"github.com/infrahq/infra/internal/server/models"
+	"github.com/infrahq/infra/uid"
+)
+
+func TestDestinationCredentials(t *testing.T) {
+	srv := setupServer(t, withAdminUser, withMultiOrgEnabled)
+	db := srv.db
+	routes := srv.GenerateRoutes()
+	destName := "foo"
+	wg := sync.WaitGroup{}
+	wg.Add(1)
```

Comment:
This use of wait group and `t.Parallel` is surprising.  I think this test would be more obvious if we modeled it like a real-world use case.

https://pkg.go.dev/golang.org/x/sync/errgroup makes managing the errors from goroutines much easier, and wraps a `WaitGroup`, so it also manages waiting for them to exit.

This test could start a goroutine that acts as the connector using `errgroup` first (I guess that's what's in the ListDestinationCredentials test case). Then run create in the main goroutine, and at the end call `assert.NilError(t, group.Wait())`. 

If you set a deadline on the context, it should prevent the goroutine from running too long in the case of failure. 

I think that would make it more clear that this is testing all 3 endpoints together. We'll also want separate test cases for different response codes, and to test the full response body. Testing only the response struct that was decoded makes it easy to miss fields in the test coverage. So we will want a table test for each of the endpoints as well eventually.

Reply:
It's elegant as is without adding a ton of complexity. Table tests here are not currently needed, but we can always add this in the future. I added a comment for clarity.

Reply:
This is abusing `t.Parallel` for something it was never designed for.